src/web3/contracts.ts

```diff
@@ -109,6 +120,32 @@ async function waitForContractKit(tries: number) {
   return contractKit
 }
 
+// This code assumes that the account for walletAddress already exists in the Keychain
+// which is a responsibility currently handled by KeychainWallet
+export function* getViemWallet(chain: Chain) {
```

Comment:
`getViemWallet` appears to require the user's account to be unlocked if this is the first time the Viem wallet is being fetched -- it doesn't appear that this is called during wallet initialization (will it eventually be?), so if there are places in the app that use the wallet but don't immediately require unlocked access, this will "greedily" request the user's PIN, right? I'm not immediately sure of places like that (maybe initializing FC SDKs?) and I imagine we'll eventually initialize Viem wallets along with our regular wallet (guaranteeing that the wallet is already unlocked), so perhaps this is a moot point.

Reply:
Very good observation. Its correct that this component greedily unlocks. Satish and I were also thinking about places that the we need a wallet without unlocking and couldn't think of any. 

We also found a pretty easy path to decouple this in the future if we ever want to initialize the wallet without unlocking, so I decided to keep it this way for now and not over-engineer. We can always change it later if we cross that bridge